### PR TITLE
Removes toString->parse roundtrip for missing value in terms agg

### LINKED ISSUE #67197: "Unparseable number" exception when missing is used with format in terms agg
Reproduces with both 7.x and master:

```
DELETE test?ignore_unavailable=true

POST test/_bulk?refresh
{"index":{}}
{"val": 1}


POST test/_search
{
  "size": 10, 
  "aggs": {
    "test": {
      "terms": {
        "field": "val",
        "format": "test ###.0#",
        "size": 10,
        "missing": 42
      }
    }
  }
}
```

The error:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "parse_exception",
        "reason" : "parse_exception: Unparseable number: \"42\""
      }
    ],
    "type" : "search_phase_execution_exception",
    "reason" : "all shards failed",
    "phase" : "query",
    "grouped" : true,
    "failed_shards" : [
      {
        "shard" : 0,
        "index" : "test",
        "node" : "0inRi1i0RIiutV9RuJiNlg",
        "reason" : {
          "type" : "runtime_exception",
          "reason" : "runtime_exception: java.text.ParseException: Unparseable number: \"42\"",
          "caused_by" : {
            "type" : "parse_exception",
            "reason" : "parse_exception: Unparseable number: \"42\""
          }
        }
      }
    ],
    "caused_by" : {
      "type" : "parse_exception",
      "reason" : "parse_exception: Unparseable number: \"42\""
    }
  },
  "status" : 500
}
```

### PR DESCRIPTION
**Pull request by imotov**
Removes unnecessary and confusing roundtrip that missing values are undertaking
in the terms agg if they are already specified as numbers. It also improves a
somewhat confusing error message when the missing value cannot be parsed
according to the specified format.

Closes #67197

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java [hunk 1]
```diff
@@ -454,7 +454,7 @@ public double parseDouble(String value, boolean roundUp, LongSupplier now) {
             try {
                 n = format.parse(value);
             } catch (ParseException e) {
-                throw new RuntimeException(e);
+                throw new RuntimeException("Cannot parse the value [" + value + "] using the pattern [" + pattern + "]");
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/DocValueFormat.java:457]: Maybe add `, e` just so we don't lose it?

**Review by nik9000 — COMMENTED** [review]
Makes me happy.

File: server/src/main/java/org/elasticsearch/search/DocValueFormat.java (hunk 1 above)
Comment (imotov) [reply server/src/main/java/org/elasticsearch/search/DocValueFormat.java:457]: Oh, good catch!

**Review by not-napoleon — APPROVED** [review]
LGTM

**Pull request merged**